### ISSUE TITLE
Plexdrive crash

### ISSUE THREAD

**Issue by breezytm** [dweidenfeld/plexdrive#320]
UGH, has anyone figured out how to keep the drive from disconnected and return that "transport endpoint is not connected" error message. 

I just created a script to monitor the drive and unmount and mount it every time that happens. Let's see how that goes. 
`#!/bin/bash
# Validate if the plexdrive mounts are available every 10 minutes

if mountpoint -q /mnt/plexdrive5
then
        echo "Google Drive is online"
else
        echo "Google Drive is offline. Mounting the volume..."
        fusermount -uz /mnt/plexdrive5 >> /dev/null 2>&1
        /usr/local/bin/plexdrive mount -c /root/.plexdrive -o allow_other /mnt/p                                                         lexdrive5 >> /dev/null 2>&1
fi
`
I am open to a better suggestion. I don't even know if this works since it hasn't crashed since I implemented it.

**Comment by felixbuenemann** [comment]
If your system uses systemd, then systemd can automatically restart plexdrive if it exits:

```ini
[Unit]
Description=Plexdrive FUSE mount
Documentation=https://github.com/dweidenfeld/plexdrive/
AssertPathIsDirectory=/mnt/plexdrive
After=network-online.target

[Service]
User=plex
Group=plex
Type=simple
ExecStart=/usr/local/bin/plexdrive mount -v 2 -o allow_other,read_only --chunk-check-threads=8 --chunk-load-threads=8 --chunk-load-ahead=2 --max-chunks=16 /mnt/plexdrive
ExecStop=/bin/fusermount -uz /mnt/plexdrive
Restart=on-abort

[Install]
WantedBy=default.target
WantedBy=plexmediaserver.service
```

However if this happens frequently you should probably try to find out why it happens.

**Comment by breezytm** [comment]
Great. Thanks. Can i use the same syntax for plexdrive v4
I use them both. V4 & 5.

V5 doesn't seem to work with Emby so I had to install V4 as well since I ran both servers on my box.

**Comment by felixbuenemann** [comment]
I have never used plexdrive v4, so maybe the commandline parameters in ExecStart are different. But other than that it should work.

**Comment by breezytm** [comment]
Yeah that's the syntax i was referring to. But hey thanks. I made the change to the plexdrive v5 service. i'll monitor it and see if it plex shows anything as unavailable.

**Comment by zmike808** [comment]
> Great. Thanks. Can i use the same syntax for plexdrive v4
> I use them both. V4 & 5.
> 
> V5 doesn't seem to work with Emby so I had to install V4 as well since I ran both servers on my box.

Plexdrive5 doesn't work with emby? That isn't possible since I tried to setup an emby server a while ago along with my plex server (while I was already using plexdrive for 90% of my media that I host), and I had 0 issues. Besides the fact that this was maybe 6 months to a year or so ago and emby was a complete PoS compared to plex and took forever to scan, albeit it did finish, and the admin and user UI were both complete trash. So, I gave up on it. I did like the extra control, but most of my users can barely get themselves setup with plex, even after I made a system for them that only requires them to create an account. And then I automagically handle the friend request acceptance for them.... 

Also, why are you mounting as readonly? there's no reason to do that. If your using an FS overlay (#MergerFSIsBestFS) you're going to want to be able to do renames/moves/deletes to keep the state consistent between the local drive and your plexdrive. 

TL;DR: If you add `--umask=0777` to your plexdrive mount, probably will solve all your problems for why version 5 didn't work.
I also use `-o allow_other,allow_non_empty_mount,default_permissions` for the fuse options, although only thing that could possible matter from those is the default_permissions, which is kinda irrelevant if you use the umask I mentioned.
